src/services/factories/tokenList.ts

```diff
@@ -2,15 +2,25 @@ import { TokenList } from 'api/tokenList/TokenListApi'
 import { SubscriptionCallback } from 'api/tokenList/Subscriptions'
 import { ExchangeApi } from 'api/exchange/ExchangeApi'
 import { TokenDetails, Command } from 'types'
-import { logDebug } from 'utils'
+import { logDebug, retry } from 'utils'
 
-export function getTokensFactory(factoryParams: {
-  tokenListApi: TokenList
-  exchangeApi: ExchangeApi
-}): (tokenId: number) => TokenDetails[] {
+import { TokenFromErc20Params, TokenFromErc20 } from './'
+
+interface Injects {
+  getTokenFromErc20(params: TokenFromErc20Params): Promise<TokenFromErc20>
+}
+
+export function getTokensFactory(
+  factoryParams: {
+    tokenListApi: TokenList
+    exchangeApi: ExchangeApi
+  },
+  injects: Injects,
+): (tokenId: number) => TokenDetails[] {
   const { tokenListApi, exchangeApi } = factoryParams
+  const { getTokenFromErc20 } = injects
 
-  const updatedIdsForNetwork = new Set<number>()
+  const updatedTokensForNetwork = new Set<number>()
```

Comment:
The name made confused me a bit at the beginning. I was thinking it was a list of tokens. 
Although a `Set` is enough, I find sth like `areTokensUpdated = new Map<Network, boolean>` a bit easier to read

Reply:
Yes, this is something I wasn't happy about. Although is redundant, makes sense to have a Map for clarity.

Reply:
Actually, can't have myself to use Map here, Set is enough. 
But, I did rename and added a comment to make the intent more clear.